### DISCUSSION TITLE
Less restrictions due to Yomichan's validation time

### DISCUSSION THREAD

**Discussion by parfait8566** [Jitendex/Jitendex#17] [Ideas]
Seems like Yomitan will considerably improve validation time: https://github.com/themoeway/yomitan/pull/304

so you could go wild or something? lol (I'd personally like more tables, adding search-only terms would be nice as well)

**Comment by stephenmk** [comment]
I've been [adding quite a few search-only tags](https://github.com/JMdictProject/JMdictIssues/issues/102) specifically to avoid having them displayed in tables.

It would maybe be possible to add an expandable list of search keys to the bottom similar to how [WWWJDIC handles them](https://user-images.githubusercontent.com/8003332/280384367-0d538575-b8e2-4278-9b61-ceff8ae29b6b.png). These sorts of expandable lists aren't supported by any version of yomichan, though. Support for HTML `summary` and `details` tags would need to be added.

Generally I don't think it's useful to display the search keys. They're hidden because it's not worth anyone's time to look at them. I think adding them in would just be distracting.

It would be useful to know when search keys are also headwords in other popular dictionaries. I'm hoping to tackle this problem with my ["jiten index" idea](https://github.com/stephenmk/Jitendex#planned-features).

**Comment by parfait8566** [comment]
I see.

One thing: Jitendex doesn't seem to show why the term is irregular or outdated (`io`, `oK`, maybe there are other cases). It'd be nice to see through a tag if the term you looked up is search-only.

**Reply by stephenmk** [reply]
> Jitendex doesn't seem to show why the term is irregular or outdated (io, oK, maybe there are other cases).

This is by design. Usually you can easily tell why it's irregular by comparing it with the other forms in the table. I guess I could include the specific reasons in hover text in the table icons. I'm not entirely sure if this would be worth the effort.

> It'd be nice to see through a tag if the term you looked up is search-only.

Irregular forms have the "irregular-form" tag, for example.

![sotsunaku](https://github.com/stephenmk/Jitendex/assets/8003332/922913e0-c8c4-4951-8bbc-06d0066e1e49)

Search only forms just don't have any tag. They also don't have reading / furigana info in the yomichan headword area.

![noboribata](https://github.com/stephenmk/Jitendex/assets/8003332/b6ceba64-932f-4c3b-8608-0217b6003946)

I could add a "search-only-form" tag similar to the "irregular-form" tag. But I'm not convinced this would be beneficial. Part of my design philosophy with this project is to try to avoid overloading users with information that they don't need.
